swift/ql/lib/codeql/swift/elements/decl/MethodDecl.qll

```diff
@@ -29,17 +11,7 @@ class MethodDecl extends AbstractFunctionDecl {
   cached
   predicate hasQualifiedName(string typeName, string funcName) {
     this.getName() = funcName and
-    (
-      exists(NominalTypeDecl c |
-        c.getFullName() = typeName and
-        c.getAMember() = this
-      )
-      or
-      exists(ExtensionDecl e |
-        e.getExtendedTypeDecl().getFullName() = typeName and
-        e.getAMember() = this
-      )
-    )
+    this.getDeclaringTypeDecl().getFullName() = typeName
```

Comment:
I do like this, having `getDeclaringDecl` and `getDeclaringTypeDecl` both widely available and clearly defined is a big plus.